### DISCUSSION TITLE
How to identify which operations can operate with images opened in sequential mode ?

### DISCUSSION THREAD

**Discussion by kipcole9** [libvips/libvips#3963] [Help wanted]
I understand the principle, but it's not always easy to work out if an operation or composition of operations can work with an image opened in sequential mode. 

Is there some documentation of which operations will work in sequential mode, or are heuristics and experimentation the order of the day?

Today with the Elixir [Image](https://github.com/elixir-image/image) library I leave the decision in the hands of the developer. But I'm working on automated imaging pipelines for web applications so I will be able to force an operation order and hoping as a result I can determine in advance if I can open the image in sequential mode.

Thanks for any guidance.

**Comment by jcupitt** [comment]
Hi @kipcole9,

I think you need the `VIPS_OPERATION_SEQUENTIAL` flag, see:

https://www.libvips.org/API/current/VipsOperation.html#vips-operation-get-flags

It's what the vips CLI uses to decide whether to open images in seq mode. `vips` will print the flag value with eg.:

```
$ vips add
add two images
usage:
   add left right out [--option-name option-value ...]
where:
   left         - Left-hand image argument, input VipsImage
   right        - Right-hand image argument, input VipsImage
   out          - Output image, output VipsImage
operation flags: sequential 
```

You could AND all flags in a pipeline. 

It's not 100% though --- whether or not something will work depends on a lot of factors.
